File: backend/transactions/device_types/update_device_type_transaction.py
from schemas.device_type_schemas import UpdateDeviceTypeRequest
from device_types.update_device_type import update_device_type
from supabase import Client

def update_device_type_transaction(
                                    client:Client,
                                    device_type: UpdateDeviceTypeRequest,
                                    hospital_id:str
                                  ):
    print("update_device_type_transaction")

    return update_device_type(
                              client, 
                              device_type,
                              hospital_id
                            )

File: frontend/app/api/transactions/deviceTypes/updateDeviceTypeTransaction.ts
import { API_BASE_URL } from "../../client/apiClient"
import { UpdateDeviceTypeFrontType } from "../../../types/deviceTypeTypes"
import { getDeviceTypesFromApi } from "../../deviceTypes/fetchDeviceTypes"
import {
         normalizeDeviceType,
         toUpdateDeviceTypeRequest
       } from "../../../utils/deviceTypeMapper"
import { authFetch } from "../../client/apiClient"

type UpdateDeviceTypeTransactionParams = {
                                            deviceType: UpdateDeviceTypeFrontType
                                            setDeviceTypes: any
                                          }

export async function updateDeviceTypeTransaction({
                                                    deviceType,
                                                    setDeviceTypes
                                                  }: UpdateDeviceTypeTransactionParams
                                                )
{
  console.log("updateDeviceTypeTransaction")

  await fetch(
                `${API_BASE_URL}/update-device-type`,
                {
                  method: "POST",
                  headers: {
                            "Content-Type":
                            "application/json"
                            },credentials: "include",
                  body: JSON.stringify(
                                          toUpdateDeviceTypeRequest(
                                                                      deviceType
                                                                    )
                                        )
                }
              )

  const deviceTypes =
    await getDeviceTypesFromApi()

  setDeviceTypes(
                   deviceTypes.map(
                                     normalizeDeviceType
                                   )
                 )
}